src/java/org/apache/cassandra/replication/ImmutableCoordinatorLogOffsets.java

```diff
@@ -112,6 +125,9 @@ public Builder()
         public Builder(int size)
         {
             this.ids = new Long2ObjectHashMap<>(size, 0.9f, false);
+
+            // Transfers are very rare, opt to save memory
+            this.transfers = new ArrayList<>(1);
```

Comment:
_technically_ using null here and only instantiating an object if a transfer is added would save the most memory